docs/sdk-writer-guide.md

```diff
@@ -13,25 +13,33 @@ These SDK writer output contains just a carbondata and carbonindex files. No met
  
  import org.apache.carbondata.common.exceptions.sql.InvalidLoadOptionException;
  import org.apache.carbondata.core.metadata.datatype.DataTypes;
+ import org.apache.carbondata.core.util.CarbonProperties;
  import org.apache.carbondata.sdk.file.CarbonWriter;
  import org.apache.carbondata.sdk.file.CarbonWriterBuilder;
  import org.apache.carbondata.sdk.file.Field;
  import org.apache.carbondata.sdk.file.Schema;
  
  public class TestSdk {
- 
+
+   // pass true or false while executing the main to use offheap memory or not
    public static void main(String[] args) throws IOException, InvalidLoadOptionException {
-     testSdkWriter();
+     if(args[0] != null) {
+       testSdkWriter(args[0]);
+     } else {
+       testSdkWriter("true");
+     }
    }
  
-   public static void testSdkWriter() throws IOException, InvalidLoadOptionException {
+   public static void testSdkWriter(String enableOffheap) throws IOException, InvalidLoadOptionException {
      String path = "/home/root1/Documents/ab/temp";
  
      Field[] fields = new Field[2];
      fields[0] = new Field("name", DataTypes.STRING);
      fields[1] = new Field("age", DataTypes.INT);
  
      Schema schema = new Schema(fields);
+
+     CarbonProperties.getInstance().addProperty("enable.offheap.sort", enableOffheap);
  
      CarbonWriterBuilder builder = CarbonWriter.builder().withSchema(schema).outputPath(path);
```

Comment:
Can you optimize the write build process?  some methods have been changed, like withSchema and some.

Reply:
it  may be failing because of invalid path given in testSdkWriter().
![image](https://user-images.githubusercontent.com/14244942/40309877-b1f88696-5d28-11e8-83c5-912de1aae70f.png)